crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -122,17 +129,45 @@ impl AuthorityStorePruner {
         total_pruned
     }
 
+    fn handle_checkpoint(
+        checkpoint_execution_state: CheckpointExecutionState,
+        objects: &DBMap<ObjectKey, Object>,
+    ) -> anyhow::Result<usize> {
+        let mut pruned = 0;
+        let mut wb = objects.batch();
+        let mut updates = HashMap::new();
+
+        for effects in checkpoint_execution_state.effects {
+            for (object_id, seq_number) in effects.modified_at_versions {
+                updates
+                    .entry(object_id)
+                    .and_modify(|version| *version = max(*version, seq_number))
+                    .or_insert(seq_number);
+            }
+        }
+        for (object_id, version) in updates {
+            let object_key = ObjectKey(object_id, version);
+            let iter = objects.iter().skip_prior_to(&object_key)?.reverse();
+            let mut start_range = object_key;
+            let end_range = ObjectKey(object_key.0, SequenceNumber::from(object_key.1.value() + 1));
+            for (key, _) in iter.take_while(|(key, _)| key.0 == object_key.0) {
+                start_range = key;
+                pruned += 1;
+            }
+            wb = wb.delete_range(objects, &start_range, &end_range)?;
```

Comment:
We are going to be adding a range tombstone for every single version of an object. Hopefully we will not see any read performance impact given we ignore range deletes but jfyi.